resolver/resolver.go

```diff
@@ -25,27 +25,47 @@ import (
 
 // Resolver holds configuration state and the resource records
 type Resolver struct {
-	masters []string
-	version string
-	config  records.Config
-	rs      *records.RecordGenerator
-	rsLock  sync.RWMutex
-	rng     *rand.Rand
-	fwd     exchanger.Forwarder
+	masters    []string
+	version    string
+	config     records.Config
+	rs         *records.RecordGenerator
+	rsLock     sync.RWMutex
+	rng        *rand.Rand
+	fwd        exchanger.Forwarder
+	httpClient *http.Client
+}
+
+func newHTTPClient(httpTimeout time.Duration) *http.Client {
```

Comment:
Talked to @spahl. He thinks our changes make sense, and he's going to wait till we merge to master (soon), and then he can rebase atop us.